service/javadsl/api/src/main/scala/com/lightbend/lagom/internal/api/ServiceReader.scala

```diff
@@ -255,17 +255,21 @@ object ServiceReader {
     override def invoke(proxy: scala.Any, method: Method, args: Array[AnyRef]): AnyRef = {
       // If it's a default method, invoke it
       if (method.isDefault) {
-        // This is the way to invoke default methods via reflection, using the JDK7 method handles API
-        val declaringClass = method.getDeclaringClass
-        // We create a MethodHandles.Lookup that is allowed to look up private methods
-        methodHandlesLookupConstructor.newInstance(declaringClass, new Integer(MethodHandles.Lookup.PRIVATE))
-          // Now using unreflect special, we get the default method from the declaring class, rather than the proxy
-          .unreflectSpecial(method, declaringClass)
-          // We bind to the proxy so that we end up invoking on the proxy
-          .bindTo(proxy)
-          // And now we actually invoke it
-          .invokeWithArguments(args: _*)
-
+        if (java.lang.reflect.Modifier.isPublic(serviceInterface.getModifiers)) {
```

Comment:
Got distracted by the verification in case the Service is defined on a scala trait. Will ignore verifications in case of scala Service API for the time being but we'll need to review when merging with scala API.